config/opal_check_withdir.m4

```diff
@@ -19,17 +20,21 @@ dnl
 # ----------------------------------------------------
 AC_DEFUN([OPAL_CHECK_WITHDIR],[
     AC_MSG_CHECKING([--with-$1 value])
-    AS_IF([test "$2" = "yes" || test "$2" = "no" || test "x$2" = "x"],
-          [AC_MSG_RESULT([simple ok (unspecified)])],
-          [AS_IF([test ! -d "$2"],
-                 [AC_MSG_RESULT([not found])
-                  AC_MSG_WARN([Directory $2 not found])
-                  AC_MSG_ERROR([Cannot continue])],
-                 [AS_IF([test "x`ls $2/$3 2> /dev/null`" = "x"],
+    AS_IF([test "$2" = "no" ],
+          [AC_MSG_RESULT([simple no (specified --without-$1)])],
+          [AS_IF([test "$2" = "yes" || test "x$2" = "x"],
+                 [AC_MSG_RESULT([simple ok (unspecified value)])],
+                 [AS_IF([test ! -d "$2"],
                         [AC_MSG_RESULT([not found])
-                         AC_MSG_WARN([Expected file $2/$3 not found])
+                         AC_MSG_WARN([Directory $2 not found])
                          AC_MSG_ERROR([Cannot continue])],
-                        [AC_MSG_RESULT([sanity check ok ($2)])]
+                        [AS_IF([test "x`ls $2/$3 2> /dev/null`" = "x"],
```

Comment:
would ```test ! -e "$2/$3"``` do the trick here ?

Reply:
Possibly. I didn't want to change the prior behavior, so I left it as is. The git diff shows the indentation change in an odd way :/

Reply:
Can't remember why we initially went with `ls` instead of `test -e`.  Seems like a (slightly) odd choice, but `ls` works just fine.